### DISCUSSION TITLE
[Account Issues]: Testing Account Access | Create Account Through App, Not Website

### DISCUSSION THREAD

**Discussion by pbaerthel** [jjjonesjr33/petlibro#66] [General]
### Device model name

Petlibro Dockstream

### Device model number

PLWF105

### Device type

Fountain

### Feature name

Europe or Germany Support

### Describe the feature

Sadly ist only Isabel with US Accounts. But I’m a German with German Petlibro account 

### What problem does this feature solve?

Add Germany server 

### Available to help with the feature development

- [X] I agree to help with testing or providing relevant information for this feature request

**Comment by jjjonesjr33** [comment]
@pbaerthel at the moment we only have the information for the US region. If you are able to provide the information for your regions API for us to test/add it that would help get the ball rolling on region support.

Please let me know if you are able to assist with this.

P.S. As of right now the workaround is to create a us account and migrate your devices to it.

**Comment by peterjeschke** [comment]
FIY I'm using a polar feeder from Germany and it's using the US API even with a German account

**Comment by RolfKoenders** [comment]
I thought the exact same thing, but I can login with my EU account (Netherlands) with the US server. I also am not able to find any other api host for any other region.

**Comment by jjjonesjr33** [comment]
@peterjeschke with your account, how was that setup originally? When I go to `https://de.petlibro.com/` it says invalid when I try to sign in. Are you able to sign into both the US site `https://petlibro.com/` and the Germany site `https://de.petlibro.com/`?

**Comment by jjjonesjr33** [comment]
@RolfKoenders, I’m curious if logging into the API works universally for any account type, regardless of the website or region it was created on. If that’s the case, it raises the question of why they don’t block creating duplicate accounts with the same information across their various websites.

From my testing, I can create accounts on any of the regional domains for Petlibro and update the region/location in the account settings. However, sharing devices across different regions/locations doesn’t seem to work.

It seems like further investigation may be necessary to determine exactly what works and where.

**Comment by peterjeschke** [comment]
I created my account using the app and selected "Germany" as my country of origin. The same credentials don't work for the website, neither petlibro.com nor de.petlibro.com.

**Comment by peterjeschke** [comment]
I was able to sign up at de.petlibro.com with the same email I used for the app _again_. I guess the shop/website accounts are completely separate from the app accounts.

**Comment by Denox78** [comment]
Hi everyone,

I'm from France and just ordered my first product

Should I create an account and register my product in the US region?

**Comment by peterjeschke** [comment]
> Hi everyone,
> 
> I'm from France and just ordered my first product
> 
> Should I create an account and register my product in the US region?

I created my account in the app and chose Germany, you should be fine choosing France

**Comment by Denox78** [comment]
> > Hi everyone,
> > I'm from France and just ordered my first product
> > Should I create an account and register my product in the US region?
> 
> I created my account in the app and chose Germany, you should be fine choosing France

Ok that's great, so you can see all your devices once added on the German's account ?

I'm also wondering if it's possible to delete all devices from an account to activate them on another (just in case I noticed everything's not working as expected)

**Comment by jjjonesjr33** [comment]
From what I’ve noticed, creating an account on the app is completely separate from their web domains. It looks like no matter which region you pick, as long as you create the account through the app, it lets you access the API.

That said, it might be worth testing a bit more to see if accounts from any region on the app work the same way.

**Comment by peterjeschke** [comment]
> > > Hi everyone,
> > > I'm from France and just ordered my first product
> > > Should I create an account and register my product in the US region?
> > 
> > I created my account in the app and chose Germany, you should be fine choosing France
> 
> Ok that's great, so you can see all your devices once added on the German's account ?

Yes, all devices are visible and shared between three accounts (all in the same region)

> I'm also wondering if it's possible to delete all devices from an account to activate them on another (just in case I noticed everything's not working as expected)

Haven't tried that

**Comment by peterjeschke** [comment]
I did some basic reverse engineering to search for host names in the Polar Feeder firmware and app to figure out if there are different endpoints for non-US scenarios. I couldn't find any in the firmware (either I didn't do it right or it's not stored in the firmware blob - maybe it's synced during the setup with the app).

But here are the (interesting) hostnames I could find:

```
mqtt.dl-aiot.com
m.designlibro.com.cn
api.designlibro.com.cn
api.us.petlibro.com
mqtt.us.petlibro.com
us.petlibro.com
```
dl-aiot.com redirects to petlibro.com.
I also found what looks to be fragments of host names:
```
-svc.dl-aiot.com
-api.dl-aiot.com
-m.dl-aiot.com
```
But domain names can't start with a hyphen, so something's missing. With some basic subdomain enumeration, I could verify that these subdomains _used to_ exist, but aren't available anymore:

```
test.m.dl-aiot.com
sit.m.dl-aiot.com
test.api.dl-aiot.com
```

Maybe the hyphen is just in there by mistake.

Looking in my DNS logs for dl-aiot.com, I can also find these host names:

```
us-mqtt-0.dl-aiot.com
oss.dl-aiot.com
```

All of this suggests that there is only one region (US) for the app.

**Comment by mavrster** [comment]
My original account is in New Zealand region and I get the login error

> File "/config/custom_components/petlibro/api.py", line 107, in request
>     raise PetLibroAPIError(f"Code: {data.get('code')}, Message: {data.get('msg')}")
> custom_components.petlibro.exceptions.PetLibroAPIError: Code: 1101, Message: MEMBER_NOT_EXIST

I created a new account in US region and got the same error.

**Reply by raphus** [reply]
i am having the same error & issue. just created a new account, and cannot login through HA

**Comment by bigsteve76** [comment]
I have a US account that I created through the app. When ever I enable the HA integration, the app on my phone will log out. You are not able to reliably log in to the app or see any of the devices. Once I disable the integration, the app become functional again. Any suggestions?

**Comment by jjjonesjr33** [comment]
@bigsteve76 you need two accounts. One which is your main for your mobile device, and another that you share your deivces to so home assistant can use it. Please review the [wiki](https://github.com/jjjonesjr33/petlibro/wiki/PetLibro-Account-Management).

**Reply by justindemuro** [reply]
So I totally made the same mistake of adding the integration to HA (complete with my sole account) before noticing the admittedly well-illuminated warning to create a secondary account to maintain simultaneous integration and app access. 

I realize what I need now, but not sure how to do this. Presumably:
1. Disable PetLibro integration
2. Log in to mobile app - create secondary account (with as much privilege as primary account as necessary for integration)
**3. Change login details of PetLibro integration to secondary account**
4. Re-enable PetLibro integration and hopefully everything works great

However, step 3 eludes me. Is there a way to update the login info in the integration without blowing it away and redeploying it?

**Comment by devbrsa** [comment]
I'm experiencing authentication problems with multiple Petlibro accounts across different regions and apps.
EU Account (Petlibro Lite):

I have an existing European account on Petlibro Lite
This account was created some time ago
Even after resetting to a simple password, I still receive error 1102: ILLEGAL_PASSWORD
I cannot log into this account using the Petlibro App (the standard app, not Lite)

US Account (New):

I created a new account in the US region
Device sharing from EU to US is not allowed (likely due to EU regulations)
Even with this fresh US account, I'm still getting the same 1102: ILLEGAL_PASSWORD error

**Testing**:
I've been using Postman to test different API endpoints (US, EU, international) with both accounts, but the results are consistently the same - 1102: ILLEGAL_PASSWORD for all combinations.

**Note**: I understand this isn't official Petlibro support, but I'd appreciate any comments from anyone who has experienced similar behavior or knows if Petlibro Lite and Petlibro App use different API systems. Could this be why neither my EU Lite account nor my new US account works with the Home Assistant integration?

**Reply by jjjonesjr33** [reply]
@devbrsa Petlibro Lite accounts are not supported, as the devices associated with Petlibro Lite are primarily Tuya-based.

**Reply by devbrsa** [reply]
@jjjonesjr33 thanks for your answer, problem I can't connect the device (PLAF203) using the main app, I'm writing messages to them, let's see.

**Comment by FFeiner98** [comment]
I am wondering if it is already possible to set it up with an EU account. As HA only shows US option, and I read above that you cannot mix US account and EU account (which contains the device)
